src/interiorpointsolver.jl

```diff
@@ -362,15 +383,15 @@ function Solver(nlp::NonlinearProgram;
         DenseKKTSystem{Float64, VT, MT}(nlp, ind_cons)
     end
 
-    xl = [nlp.xl;view(nlp.gl,ind_cons.ind_ineq)]
-    xu = [nlp.xu;view(nlp.gu,ind_cons.ind_ineq)]
-    x = [nlp.x;zeros(ns)]
-    l = nlp.l
-    zl= [nlp.zl;zeros(ns)]
-    zu= [nlp.zu;zeros(ns)]
+    xl = [get_lvar(nlp);view(get_lcon(nlp),ind_cons.ind_ineq)]
```

Comment:
Here I would maybe add a few type hinting for the compiler. Like: 
```julia
xl_lvar = get_lvar(nlp)::Vector{Float64}
xl_lcon = view(get_lcon(nlp)::Vector{Float64},ind_cons.ind_ineq)
xl =Float64[xl_lvar; xl_lcon]
```
(but might be a bit ugly)

Reply:
Here, we're not certain that `get_lvar(nlp)` is `Vector{Float64}`, since we're taking `AbstractNLPModel` as an input. One possibility is to write a function e.g., `function get_initial_vectors` and do specialization for each NLPModel type. But I don't think this would be that much performance-critical